generate_new_lesions/augment_lesions_intra_dataset.py

```diff
@@ -0,0 +1,495 @@
+    new_sc = im_healthy_sc.copy()
+
+    # Get centerline from healthy SC seg. The centerline is used to project the lesion from the pathological image
+    healthy_centerline = get_centerline(im_healthy_sc_data)
+
+    # Initialize numpy arrays with the same shape as the healthy image once
+    im_augmented_data = np.copy(im_healthy_data)
+    im_augmented_lesion_data = np.zeros_like(im_healthy_data)
+
+    # Select random coordinate on the centerline
+    # index is used to have different seed for every subject to have different lesion positions across different subjects
+    rng = np.random.default_rng(args.seed + index)
+
+    for i, patho_lesion_data in enumerate(extracted_patho_lesions):
+        print(f"Inserting Lesion {i+1} out of {len(extracted_patho_lesions)}")
+        """
+        Get intensity ratio between healthy and pathological SC and normalize images.
+        The ratio is used to multiply the lesion in the healthy image.
+        """    
+        # Extract SC mask in the neighbourhood of the lesion
+        patho_lesion_data_dilated = binary_dilation(patho_lesion_data, structure=generate_binary_structure(3, 5), iterations=3)
+        im_patho_sc_dil_data = im_patho_sc_data * patho_lesion_data_dilated
+        
+        # compute the ratio of intensities between lesion and SC in the pathological image
+        lesion_sc_ratio_patho = np.mean(im_patho_data[patho_lesion_data > 0]) / np.mean(im_patho_data[(im_patho_sc_dil_data > 0) & (patho_lesion_data == 0)])    # without lesion
+        print(f"Mean lesion/SC Intensity Ratio of Patho Subject {sub_patho}: {lesion_sc_ratio_patho}")
+        # Make sure the intensity ratio is always < 1 (i.e. the lesion is always darker (because mp2rage) than the healthy SC)
+        if lesion_sc_ratio_patho > 1:
+            lesion_sc_ratio_patho = 1 / lesion_sc_ratio_patho
+            print(f"Mean lesion/SC Intensity Ratio of Patho Subject {sub_patho} (after inversion): {lesion_sc_ratio_patho}")
+
+        # Create 3D bounding box around non-zero pixels (i.e., around the lesion)
+        lesion_coords = np.argwhere(patho_lesion_data > 0)
+
+        # NOTE: This loop is required because the lesion from the original patho image could be cropped if it is going
+        # outside of the SC in the healthy image. So, the loop continues until the lesion inserted in the healthy image
+        # has similar volume w.r.t the original lesion
+        j = 0
+        while True:            
+            # Initialize numpy arrays with the same shape as the healthy image
+            im_augmented_data_new = np.copy(im_augmented_data)
+            im_augmented_lesion_data_new = np.copy(im_augmented_lesion_data)
+
+            # get the volume of the augmented lesion first
+            lesion_vol_aug_init = get_lesion_volume(im_augmented_lesion_data_new, im_augmented_lesion.dim[4:7], debug=False)
+
+            # New position for the lesion (on the centerline)
```

Comment:
```suggestion
            # Choose a random coordinate on the target centerline. This will be the position of the new lesion on the target image
```

Reply:
Fixed in commit  https://github.com/ivadomed/model_seg_sci/pull/45/commits/e2ac381f9ed44ecdad0b696273186082d920b604